### PR TITLE
[SYCL] June 2022 Release Notes

### PR DESCRIPTION
**Pull request by againull**

### CODE REVIEW

File: sycl/ReleaseNotes.md [hunk 1]
```diff
@@ -1,3 +1,386 @@
...
+  - Added support for memory intrinsics for the ESIMD_EMULATOR plugin. [1a8f501]
+  - Added support for named barrier APIs for ESIMD. [1df0038]
+  - Added support for dpas API for ESIMD. [5881938]
+  - Added support for lsc memory access APIs for ESIMD. [4bd50e7]
+  - Added support for `buffer_location` property to the `sycl::buffer`. [9808525]
+  - Added `single_task` support for ESIMD_EMULATOR backend. [2331160]
+  - Added support for SVM 1,2,4-elements gather/scatter for ESIMD. [e200720]
+  - Added support for `bf16` builtins operating on storage types for CUDA backend. [413a9ef]
+  - Added `backend_version` device property. [4b1a4bc]
+  - Added support for the `invoke_simd` feature. [4072557][8471ff3][8c7bb45][62afb59][3e1c1bf][8c7bb45]
```
Comment (asudarsa) [inline sycl/ReleaseNotes.md:60]: [8c7bb45] is repeated twice. Also, this seems like a better fit in 'Tools' section.

Comment (againull) [reply sycl/ReleaseNotes.md]: Removed duplicate. Commits implementing this feature touch many parts of the compiler - frontend, llvm passes, sycl-post-link tool etc. That's why I put it in SYCL Compiler section. Correct me if my understanding is wrong.

**Review by steffenlarsen — COMMENTED** [review]
Content look good to me. I just have some suggestions for consistency.

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Pull request merged**